rdkPlugins/Storage/source/Storage.cpp

```diff
@@ -89,6 +94,28 @@ bool Storage::preCreation()
             return false;
         }
     }
+
+#ifdef USE_MOUNT_TUNNEL
+    // Create host directory for the mount tunnel
+    if (!DobbyRdkPluginUtils::mkdirRecursive(mTempMountPointOutsideContainer, 0755))
+    {
+        AI_LOG_WARN("failed to create dir '%s'", mTempMountPointOutsideContainer.c_str());
+        return false;
+    }
+    // Create directory inside the container rootfs for the mount tunnel
+    if (!DobbyRdkPluginUtils::mkdirRecursive(mMountPointInsideContainer, 0755))
+    {
+        AI_LOG_WARN("failed to create dir '%s'", mMountPointInsideContainer.c_str());
+        return false;
+    }
+
+    if(mount(mTempMountPointOutsideContainer.c_str(), mTempMountPointOutsideContainer.c_str(), NULL, MS_BIND, NULL) != 0)
```

Comment:
Just checking - is this line right ?  It's bind mounting `mTempMountPointOutsideContainer` back to `mTempMountPointOutsideContainer` ?

Reply:
yes we need the bind mount here, otherwise mount propagation does not work.  
From: https://brauner.io/2023/02/28/mounting-into-mount-namespaces.html
```
mount --bind /opt /opt
mount --make-private /opt
mount --make-shared /opt
# Create new mount namespace with all mounts turned into dependent mounts.
unshare --mount --propagation=slave
```